pkg/cortex/modules.go

```diff
@@ -630,7 +636,9 @@ func (t *Cortex) setupModuleManager() error {
 	mm.RegisterModule(MemberlistKV, t.initMemberlistKV, modules.UserInvisibleModule)
 	mm.RegisterModule(Ring, t.initRing, modules.UserInvisibleModule)
 	mm.RegisterModule(Overrides, t.initOverrides, modules.UserInvisibleModule)
-	mm.RegisterModule(Distributor, t.initDistributor)
+	mm.RegisterModule(Distributor, nil)
+	mm.RegisterModule(DistributorAPI, t.initDistributorAPI)
+	mm.RegisterModule(DistributorService, t.initDistributorService)
```

Comment:
DistributorService should use `modules.UserInvisibleModule` option.

Reply:
Makes sense